op-chain-ops/srcmap/solutil.go

```diff
@@ -69,84 +69,107 @@ func parseInstrMapping(last InstrMapping, v string) (InstrMapping, error) {
 	return out, err
 }
 
+func loadLineColData(srcFs fs.FS, srcPath string) ([]LineCol, error) {
+	dat, err := fs.ReadFile(srcFs, srcPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read source %q: %w", srcPath, err)
+	}
+	datStr := string(dat)
+	out := make([]LineCol, len(datStr))
+	line := uint32(1)
+	lastLinePos := uint32(0)
+	for i, b := range datStr { // iterate the utf8 or the bytes?
+		col := uint32(i) - lastLinePos
+		out[i] = LineCol{Line: line, Col: col}
+		if b == '\n' {
+			lastLinePos = uint32(i)
+			line += 1
+		}
+	}
+	return out, nil
+}
+
+type SourceID uint64
+
+func (id *SourceID) UnmarshalText(data []byte) error {
+	v, err := strconv.ParseUint(string(data), 10, 64)
+	if err != nil {
+		return err
+	}
+	*id = SourceID(v)
+	return nil
+}
+
+// SourceMap is a util to map solidity deployed-bytecode positions
+// to source-file, line and column position data.
+// It is best used in combination with foundry.SourceMapFS to load the source-map.
+// The source-map functionality is tested as part of the FS.
 type SourceMap struct {
-	// source names
-	Sources []string
+	srcFs       fs.FS
+	srcIDToPath map[SourceID]string
 	// per source, source offset -> line/col
-	PosData [][]LineCol
+	// This data is lazy-loaded.
+	PosData map[SourceID][]LineCol
 	// per bytecode byte, byte index -> instr
 	Instr []InstrMapping
 }
 
-func (s *SourceMap) Info(pc uint64) (source string, line uint32, col uint32) {
+func (s *SourceMap) Info(pc uint64) (source string, line uint32, col uint32, err error) {
```

Comment:
nit: I would love some comments around this method. the short variable names make it hard to understand from context. Goes for the other functions in this file too, but this was the one I noticed it on.

Reply:
Added doc comments :+1: